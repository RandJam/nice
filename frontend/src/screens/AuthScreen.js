import React, { Component } from 'react';
import {
  StyleSheet, Text, View, Button, Image
} from 'react-native';
import RectangleButton from '../components/RectangleButton';
import axios from 'axios';

const serverUrl = 'http://10.31.2.119:5000';
const axiosMain = axios.create({
  baseURL: serverUrl
});

export default class AuthScreen extends Component {

  static navigationOptions = {
    header: 'none'
  }

  someMethod() {
    'hello'
  }

  handleLogInPress = () => {
    axiosMain.get('/')
    .then(function (response) { console.log(response.data)
    })
    .catch((err) => console.log(err))
    // this.props.navigation.navigate('HomeScreen');
  }

  render() {
    return (
      <View style={styles.wrapper}>
        <View style={styles.helloWrapper}>
          <Image
            source={require('../assets/ammar-home.png')}
            style={styles.ammar}
          />
          <Text style={styles.helloText}>Hello</Text>
          <RectangleButton
            text="Sign in with Google"
            color={'indigo'}
            backgroundColor={'blue'}
            handleOnPress={this.handleLogInPress}
          />
        </View>
      </View>
    );
  }
}


const styles = StyleSheet.create({
  wrapper: {
    flex: 1,
    display: 'flex',
  },
  helloWrapper: {
    flex: 1,
    display: 'flex',
    marginTop: 30,
    padding: 30,
  },
  ammar: {
    width: 50,
    height: 50,
    marginTop: 50,
    marginBottom: 40,
  },
  helloText: {
    fontSize: 30,
    color: 'orange',
    fontWeight: '300',
    marginBottom: 40
  }
})
