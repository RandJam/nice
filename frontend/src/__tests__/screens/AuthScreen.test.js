import React, { TextInput } from "react";
import renderer from "react-test-renderer";
import axiosMain from '../../axios';
import MockAdapter from 'axios-mock-adapter';
import AuthScreen from "../../screens/AuthScreen";

describe ("AuthScreen", () => {

  it("renders without crashing", () => {
    const rendered = renderer.create(<AuthScreen />).toJSON();
    expect(rendered).toBeTruthy();
    expect(rendered).toMatchSnapshot();
  });

  it('renders input field with placeholder', function () {
      const expectedPlaceholder = 'write something';
      expect(textInput).to.have.length(1);
      expect(textInput.props().placeholder).to.eql(expectedPlaceholder);
        });



  // it('calls navigate on pressing sign in', () => {
  //   const fakeNavigation = {navigate: jest.fn()};
  //
  //   let authScreen = renderer.create(<AuthScreen
  //     navigation={fakeNavigation}
  //     />
  //   ).getInstance();
  //
  //   authScreen.handleLogInPress();
  //   expect(fakeNavigation.navigate).toHaveBeenCalledTimes(1);
  // });
  //
  // it('shoud call componentDidMount', () => {
  //   const wrapper = shallow(<AuthScreen />);
  //   wrapper.instance().componentDidMount();
  // });

})
