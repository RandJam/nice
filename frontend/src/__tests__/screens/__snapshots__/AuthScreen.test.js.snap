// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthScreen renders without crashing 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "display": "flex",
      "flex": 1,
      "justifyContent": "center",
      "width": null,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "borderColor": "#fff",
          "borderWidth": 1,
          "margin": 20,
          "marginBottom": 0,
          "padding": 20,
          "paddingBottom": 10,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="email"
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,1)",
            "fontSize": 16,
            "height": 40,
            "marginBottom": 10,
            "padding": 10,
          }
        }
        value=""
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="password"
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,1)",
            "fontSize": 16,
            "height": 40,
            "marginBottom": 10,
            "padding": 10,
          }
        }
        value=""
      />
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "rgba(255,255,255,1)",
          "borderColor": "#fff",
          "borderWidth": 1,
          "margin": 20,
          "opacity": 1,
          "padding": 20,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Login
      </Text>
    </View>
  </View>
</View>
`;
